lib/metasploit_payloads/mettle.rb

```diff
@@ -155,5 +155,34 @@ def self.warn_local_path(path)
         @local_paths << path
       end
     end
+
+    #
+    # List extensions which are available for loading
+    #
+    def self.available_extensions(platform)
+      dir_path = path("#{platform}", 'bin')
+      if dir_path.nil?
+        full_path = ::File.join([platform, 'bin'])
+        fail RuntimeError, "#{full_path} not found", caller
+      end
+
+      extensions = ::Dir.entries(dir_path)
+      # Only return extensions!
+      extensions - [ '.', '..', 'mettle', 'mettle.bin' ]
+    end
+
+    #
+    # Load and return the contents of an extension as an object
+    #
+    def self.load_extension(platform, name, suffix = '')
+      if suffix == 'bin'
```

Comment:
Should we maybe have made the method parameter be either `:process_image` or `:exec` instead of having to infer it? Was it an issue with `.exe` extensions or something?

Reply:
I'm not sure if .exe extensions were tested, since they don't build without #105 anyway (and there is other code missing for that too).

Reply:
Hmm, I like your idea though after sleeping on it. I'll update!

Reply:
Looking more closely, load_extension doesn't even need to exist - self.read already does exactly this. Also, it seems that if you use load_extension, you end up with .bin appended twice! So, this function should just be deleted.

Reply:
welp!

Reply:
If we're going to go that far, I would create a sub class like `Mettle::Extension` that you instantiate for each extension. Maybe even a class for each type of extension. That way you could have individual configuration/validation for each binary. You could then have a `#extension` method for the Mettle payload, so that calls would look at bit like:
```ruby
MetasploitPayloads::Mettle.new('aarch64-linux-musl', config).
  extension(:sniffer).
  to_binary(:process_image)
```